### ISSUE TITLE
[java] testClassPattern no applied to class that inherits all its @Test methods

### ISSUE THREAD

**Issue by liamsharp** [pmd/pmd#4350]
**Affects PMD Version:** 
6.53.0

**Rule:**
ClassNamingConventions

Please provide the rule name and a link to the rule documentation:
https://pmd.github.io/latest/pmd_rules_apex_codestyle.html#classnamingconventions

**Description:**

If a class containing no `@Test` methods inherits from an abstract class that does have `@Test` methods it's not regarded as a test class itself and bypasses the `testClassPattern` check as part of `ClassNamingConventions`

**Code Sample demonstrating the issue:**

```java

package com.github.screamingfrog;

class Derived extends AbstractUnitTest
{

    // Not detected as a Test class so ClassNamingConventions for testClassPattern not applied
    
    @Override
    int getVal()
    {
        return 0;
    }
    

}


```

```java

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

abstract public class AbstractUnitTest 
{

    abstract int getVal();
    
    @Test
    public void testApp()
    {
        assertEquals(1, getVal());
    }
}

```

**Expected outcome:**

PMD should report this as a violation.

**Running PMD through:** *[ Maven ]*

I've created a stand along maven project to reproduce the issues here:
https://github.com/screamingfrog/hello-world-cli/tree/pmd-testclasspattern-issue
